az-digital/az_quickstart
# Allow empty end date (Closes #725)

This PR allows for the event content type to have events with no end date/time (using a zero time increment)

## Description
Currently events in QS2 require entering an end date/time, unless the date is marked as "All Day." This can present a challenge for content editors, who often need to enter dates with a defined start point but no end time.

I attended a session at Drupal Con on Smart Date where they recommended creating a duration of "Zero" to handle events with no end time. This is also mentioned in the Smart Date documentation: https://www.drupal.org/project/smart_date

For this PR, I've added the Zero duration to the field:
![image](https://user-images.githubusercontent.com/10873398/119212399-dce28880-ba6c-11eb-8ca0-a1902bd11697.png)

I also updated the AP Style display settings on each view mode of the content type (default, row, and card) to turn off "Show All Day instead of midnight (as that conflicted with displaying start time only and is not needed with the Smart Date implementation).

Composer.json has also been updated to require Smart Date 3.3.0 with this patch: https://git.drupalcode.org/project/smart_date/-/commit/a1e18189315d74d75493e3983d6962fbebb31471.patch

## Related Issue
#725 

## How Has This Been Tested?
In Probo and locally.

Probo site: https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-797.probo.build/calendar

To test:

1. Create a few events with a start time only (choose Zero as the duration)
2. Create more events with different time patterns (All Day, starting at noon, starting at midnight)
3. Verify that the time displays correctly on the nodes
3. Verify that the time displays correctly on the /calendar page
4. Verify that the time displays correctly on the sidebar block on nodes highlighting other events
5. Verify that the time appears correctly in the View paragraph for AZ Events with the grid display
![image](https://user-images.githubusercontent.com/10873398/119517370-67a4db00-bd2c-11eb-9222-bc08b5282a0b.png)


## Types of changes


## Checklist:
